apps/gitness/src/pages/pull-request/stores/pull-request-store.tsx

```diff
@@ -0,0 +1,255 @@
+import { create } from 'zustand'
+import { produce } from 'immer'
+import { mergePullReqOp, commentStatusPullReq as apiCommentStatusPullReq } from '@harnessio/code-service-client'
+import { CodeCommentState, PullRequestState } from '../types/types'
+import { PullRequestDataState } from '@harnessio/views'
+export const codeOwnersNotFoundMessage = 'CODEOWNERS file not found'
+export const codeOwnersNotFoundMessage2 = `path "CODEOWNERS" not found`
+export const codeOwnersNotFoundMessage3 = `failed to find node 'CODEOWNERS' in 'main': failed to get tree node: failed to ls file: path "CODEOWNERS" not found`
+export const oldCommitRefetchRequired = 'A newer commit is available. Only the latest commit can be merged.'
+export const prMergedRefetchRequired = 'Pull request already merged'
+export const POLLING_INTERVAL = 10000
+
+export const usePullRequestDataStore = create<PullRequestDataState>((set, get) => ({
+  repoMetadata: undefined,
+  setRepoMetadata: metadata =>
+    set(
+      produce(draft => {
+        draft.repoMetadata = metadata
+      })
```

Comment:
Yeah, mutations that surface as not being mutations are kinda nice. It's not a blocker, I'm just not a big fan of dependencies rather than simple helper functions that we control. Carry on, don't mind me lol